exchanges/okx/okx.go

```diff
@@ -0,0 +1,4441 @@
+		return nil, err
+	}
+	if len(resp) == 1 {
+		return &resp[0], nil
+	}
+	return nil, errFundingRateHistoryNotFound
+}
+
+// GetOptionMarketData retrieves option market data.
+func (ok *Okx) GetOptionMarketData(ctx context.Context, underlying string, expTime time.Time) ([]OptionMarketDataResponse, error) {
+	params := url.Values{}
+	if underlying == "" {
+		return nil, errMissingRequiredUnderlying
+	}
+	params.Set("uly", underlying)
+	if !expTime.IsZero() {
+		params.Set("expTime", fmt.Sprintf("%d%d%d", expTime.Year(), expTime.Month(), expTime.Day()))
+	}
+	var resp []OptionMarketDataResponse
+	return resp, ok.SendHTTPRequest(ctx, exchange.RestSpot, getOptionMarketDateEPL, http.MethodGet, common.EncodeURLValues(publicOptionalData, params), nil, &resp, false)
+}
+
+// GetEstimatedDeliveryPrice retrieves the estimated delivery price which will only have a return value one hour before the delivery/exercise.
+func (ok *Okx) GetEstimatedDeliveryPrice(ctx context.Context, instrumentID string) ([]DeliveryEstimatedPrice, error) {
+	var resp []DeliveryEstimatedPrice
+	params := url.Values{}
+	if instrumentID == "" {
+		return nil, errMissingRequiredParamInstID
+	}
+	params.Set("instId", instrumentID)
+	return resp, ok.SendHTTPRequest(ctx, exchange.RestSpot, getEstimatedDeliveryPriceEPL, http.MethodGet, common.EncodeURLValues(publicEstimatedPrice, params), nil, &resp, false)
+}
+
+// GetDiscountRateAndInterestFreeQuota retrieves discount rate level and interest-free quota.
+func (ok *Okx) GetDiscountRateAndInterestFreeQuota(ctx context.Context, currency string, discountLevel int8) ([]DiscountRate, error) {
+	var response []DiscountRate
+	params := url.Values{}
+	if currency != "" {
+		params.Set("ccy", currency)
+	}
+	if discountLevel > 0 && discountLevel < 5 {
+		params.Set("discountLv", strconv.Itoa(int(discountLevel)))
+	}
+	return response, ok.SendHTTPRequest(ctx, exchange.RestSpot, getDiscountRateAndInterestFreeQuotaEPL, http.MethodGet, common.EncodeURLValues(publicDiscountRate, params), nil, &response, false)
+}
+
+// GetSystemTime Retrieve API server time.
+func (ok *Okx) GetSystemTime(ctx context.Context) (*time.Time, error) {
+	var resp []ServerTime
+	if err := ok.SendHTTPRequest(ctx, exchange.RestSpot, getSystemTimeEPL, http.MethodGet, publicTime, nil, &resp, false); err != nil {
+		return nil, err
+	}
+	if len(resp) == 1 {
+		return &(resp[0].Timestamp), nil
+	}
+	return nil, errNoValidResponseFromServer
+}
+
+// GetLiquidationOrders retrieves information on liquidation orders in the last day.
+func (ok *Okx) GetLiquidationOrders(ctx context.Context, arg *LiquidationOrderRequestParams) (*LiquidationOrder, error) {
+	params := url.Values{}
+	if !(arg.InstrumentType == okxInstTypeMargin || arg.InstrumentType == okxInstTypeFutures ||
+		arg.InstrumentType == okxInstTypeSwap || arg.InstrumentType == okxInstTypeOption) {
```

Comment:
```suggestion
	if arg.InstrumentType != okxInstTypeMargin && arg.InstrumentType != okxInstTypeFutures && arg.InstrumentType != okxInstTypeSwap && arg.InstrumentType != okxInstTypeOption {
```
Please update error. What if it is populated with a different field?